runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -278,6 +299,59 @@ public void onGameTick(GameTick tick)
 				}
 			}
 		}
+		if (isInAgilityPyramid())
+		{
+			boolean foundLowerBlock = false;
```

Comment:
I think we can greatly simplify the logic in this file.

The `blocks` Map contains all of the necessary information to determine the state of the blocks, so we should be able to reduce nesting and branching and get rid of the `lastBlockCoordinate` and `foundBlock` variables entirely.

Consider something like this instead: (untested, but this should convey the idea)
```
for (NPC npc : client.getNpcs() {
    if (npc.getId() == PYRAMID_BLOCK_LOWER_ID) {
        if (npc.getWorldLocation().getX() == PYRAMID_BLOCK_LOWER_REST_X) {
            blocks.putIfAbsent(PYRAMID_BLOCK_LOWER_ID, Instant.now());
        } else {
            blocks.clear();
        }
    }
    if (npc.getId() == PYRAMID_BLOCK_UPPER_ID) {
        if (npc.getWorldLocation.getY() == PYRAMID_BLOCK_UPPER_REST_Y) {
            blocks.putIfAbsent(PYRAMID_BLOCK_UPPER_ID, Instant.now());
        } else {
            blocks.clear();
        }
    }
}
```

Then, we can also get rid of `lastLowerBlockCoordinate` and `lastUpperBlockCoordinate` everywhere.